hearth/sig/lib/hearth/telemetry.rbs

```diff
@@ -0,0 +1,54 @@
+module Hearth
+  module Telemetry
+    class TelemetryProvider
```

Comment:
This one doesn't belong here (all the rbs files should match rb directory structure).

Reply:
I was trying to consolidate but with the introduction of base classes, this is no longer needed (`TelemetryProviderBase` is replaces this)